packages/muelu/src/Graph/Containers/MueLu_Aggregates_decl.hpp

```diff
@@ -114,85 +144,133 @@ namespace MueLu {
+  
+        This has to be done by the aggregation routines.
+    */
+    void SetNumGlobalAggregates(GO nGlobalAggregates) { numGlobalAggregates_ = nGlobalAggregates; }
+
+    ///< returns the number of aggregates of the current processor. Note: could/should be renamed to GetNumLocalAggregates?
+    KOKKOS_INLINE_FUNCTION LO GetNumAggregates() const {
+      return numAggregates_;
+    }
+
     //! @brief Record whether aggregates include DOFs from other processes.
-    void AggregatesCrossProcessors(const bool &flag) {aggregatesIncludeGhosts_ = flag;};
+    KOKKOS_INLINE_FUNCTION void AggregatesCrossProcessors(const bool& flag) {
+      aggregatesIncludeGhosts_ = flag;
+    }
 
     /*! @brief Return false if and only if no aggregates include DOFs from other processes.
 
         Used in construction of tentative prolongator to skip a communication phase.
     */
-    bool AggregatesCrossProcessors() const {return aggregatesIncludeGhosts_;};
+    KOKKOS_INLINE_FUNCTION bool AggregatesCrossProcessors() const {
+      return aggregatesIncludeGhosts_;
+    }
 
     /*! @brief Returns a nonconstant vector that maps local node IDs to local aggregates IDs.
 
         For local node ID i, the corresponding vector entry v[i] is the local aggregate id to which i belongs on the current processor.
     */
-    RCP<LOMultiVector> & GetVertex2AggIdNonConst()     { return vertex2AggId_;       }
+    RCP<LOMultiVector>& GetVertex2AggIdNonConst()     { return vertex2AggId_;       }
 
-    /*! @brief Returns nonconstant vector that maps local node IDs to owning processor IDs.
+    /*! @brief Returns nonconsant vector that maps local node IDs to owning processor IDs.
 
         For local node ID i, the corresponding vector entry v[i] is the owning processor ID.
     */
-    RCP<LOVector> & GetProcWinnerNonConst()       { return procWinner_;         }
+    RCP<LOVector>& GetProcWinnerNonConst()       { return procWinner_;         }
     /*! @brief Returns constant vector that maps local node IDs to local aggregates IDs.
 
         For local node ID i, the corresponding vector entry v[i] is the local aggregate id to which i belongs on the current processor.
     */
-    const RCP<LOMultiVector> & GetVertex2AggId() const { return vertex2AggId_;       }
+    const RCP<LOMultiVector>& GetVertex2AggId() const { return vertex2AggId_;       }
 
     /*! @brief Returns constant vector that maps local node IDs to owning processor IDs.
 
         For local node ID i, the corresponding vector entry v[i] is the owning processor ID.
     */
-    const RCP<LOVector> & GetProcWinner() const   { return procWinner_;         }
+    const RCP<LOVector>& GetProcWinner() const   { return procWinner_;         }
 
     //! Returns true if node with given local node id is marked to be a root node
-    bool IsRoot(LO i) const               { return isRoot_[i];          }
+    KOKKOS_INLINE_FUNCTION
+    bool IsRoot(LO i) const                      { return isRoot_(i);          }
```

Comment:
This is used on host by the classical routines and on device by Kokkos factories, so this will most likely crash.

Reply:
I'm fairly certain this can go on host for now, seeing as all the parts of MueLu that called `IsRoot` from device were commented out anyway. We'll revisit whether that should realistically live on host/device and how the calls to it should be handled later.